### ISSUE TITLE
Can't select the DevContainer definition when using multiple definitions in one repo

### ISSUE THREAD

**Issue by OneCyrus** [microsoft/vscode-remote-release#10435]
- VSCode Version: 1.95.0
- Local OS Version: Windows 11
- Remote OS Version: Ubuntu 22.04
- Remote Extension/Connection Type: Containers

- Logs: not sure where I could find a log related to this flow/process.

Steps to Reproduce:

1. CTRL+SHIFT+P: DevContainers: Clone Repository in Container Volume
2. Select a repository with multiple DevContainer definitions
3. I would expect the prompt to select the definition. But it auto-selects the default/root devcontainer definition. This used to work perfectly before

Does this issue occur when you try this locally?: it's a feature of the remote extension
Does this issue occur when you try this locally and all extensions are disabled?: it's a feature of the remote extension

**Comment by OneCyrus** [comment]
basically this seems to be broken: https://github.com/devcontainers/spec/issues/6

**Comment by chrmarti** [comment]
Giving the choice upfront is harder in this case because the repository isn't initially cloned yet, so the configurations are not on the local machine yet.

I will fix the `F1` > `Dev Containers: Switch Container` command available while already connected to the dev container to unblock this.

**Comment by chrmarti** [comment]
This is now available in Dev Containers 0.390.0-pre-release. You still need to connect to the main dev container first, but you can then use `F1` > `Dev Containers: Switch Container` to connect to the other configurations. Please give that a try and let me know if that works for you. Thanks.

**Comment by OneCyrus** [comment]
@chrmarti I'm pretty sure this exact scenario was working before. It used to clone the repository in the bootstrap container and then was asking to select the DevContainer config. It was a deferred prompt until the extension actually knew the content of the repo.

Not sure when this actually changed/broke. I need to try again with older vscode/extension versions.

**Comment by chrmarti** [comment]
There is another prompt when there is no devcontainer.json at all that will let you pick from the list of configuration templates. That should still work. From the issue description above, you did not seem to refer to that though.

**Comment by OneCyrus** [comment]
May be I mixed it up with a remote SSH connection and then "reopen in devcontainer" where the prompt appears.

**Comment by pl4nty** [comment]
this is still broken in v0.391.0-pre-release
1. clone a new repo
2. devcontainers only searches for `.devcontainer.json` and `devcontainer/devcontainer.json`
3. prompted to create a devcontainer.json because "none exists"
4. when temp container is ready, I use `Dev Containers: Switch Container` to swap to my intended container

**Comment by chrmarti** [comment]
@pl4nty That's the current solution/workaround. If you need the prompt when connecting for the first time, please open a new feature request.
